src/shoppingListProducts/repository/productsRepository.js

```diff
@@ -0,0 +1,14 @@
+import { prismaClient } from "../../../database/prismaClient.js";
+
+async function insertProductsInShoppingList(products, id) {
+  for (const product of products) {
+    await prismaClient.shoppingListProducts.create({
+      data: {
+        name: product,
+        shoppingListId: id,
+      },
+    });
+  }
```

Comment:
you can use the `createMany` function of Prisma to do that!
would be something like:
```ts
await prismaClient. shoppingListProducts.createMany({
  data: products.map((product) => ({
    name: product.product,
    shoppingListId: product.id
  }))
})
```
Learn more [here](https://www.prisma.io/docs/concepts/components/prisma-client/crud#create-multiple-records)

Also, i don't think `products` is a good name to that variable, sounds like shoppingLists or something like that.

Reply:
createMany é uma boa

products é mais adequado do que shoppingList nesse caso visto que o conteúdo é um set de products enquanto uma shoppingList contém um set de products somado a informações adicionais, como nome da lista. 

Faria sentido chamar de shoppingList se @NicolasPereira escolhesse criar a lista e os produtos ao mesmo tempo, onde os produtos seria embedded documents. O que casaria muito bem com a sua sugestão de denormalizar a modelagem de dados.

Nesse caso, uma request tipo essa 👇 seria usada para criar a lista e os produtos
```
POST /shopping-lists

{
  “name”: “Lista foo”,
  “products”: [
    { “name”: “Arroz” },
    { “name”: “Feijão” }
  ]
}
```

Detalhe que não é necessário denormalizar a modelagem para implementar essa sugestão de rota.

Inclusive, me parece ser interessante usar essa rota sugerida para criar a lista e vários produtos de uma vez. Daí vc deixaria `/shopping-list/:id/products(/:id)` para operações individuais (se necessário — adicionar/editar/remover um produto por vez)

@NicolasPereira independente de fazer esse esquema embedded ou não, é importante passar os produtos como objeto, e não como string. Imagina que esses objetos são DTOs (pq são), só acontece que eles não tem ids ainda.

Reply:
Eu tinha pensando em usar o `createMany` só fiquei na dúvida sobre a questão do id da lista se eu conseguiria passar ele dentro do create many, visto que eu pego ele como um parametro da URL. 

Sobre a ideia que o william falou, eu tb pensei muito nisso, não gostei dessa ideia que montei de ter um array com o texto do alimento, talvez eu vá montar o array com cada item sendo "normalizado" com o atributo name.

Quis fazer o back-end mais simples possivel pois a galera que ta montando o front-end e o reconhecimento de imagem não manja tando de codigo.

Reply:
Tendi, mas relaxa tenho certeza queles conseguem mudar de `”Arroz”` para `{ name: “Arroz” }` do lado deles hahah

Reply:
Consegue passar o id sim, não interfere em nada, pode ficar tranquilo